include/triton/Conversion/TritonGPUToLLVM/Utility.h

```diff
@@ -778,13 +778,20 @@ emitBaseIndexForMfmaLayout(Location loc, RewriterBase &rewriter,
   Value laneId = urem(threadId, effectiveWarpSize);
 
   Value warpId = udiv(threadId, warpSize);
-  Value warpId0 =
-      urem(urem(warpId, warpsPerCTA[0]), i32_val(shape[0] / nonKDim));
-  Value warpId1 = urem(urem(udiv(warpId, warpsPerCTA[0]), warpsPerCTA[1]),
-                       i32_val(shape[1] / nonKDim));
-
-  Value offWarp0 = mul(warpId0, i32_val(nonKDim));
-  Value offWarp1 = mul(warpId1, i32_val(nonKDim));
+  SmallVector<Value> multiDimWarpId = delinearize(
+      rewriter, loc, warpId, _warpsPerCTA, triton::gpu::getOrder(mfmaLayout));
+  if (shape[0] >= nonKDim) {
```

Comment:
Still need to figure out why we need this and why are both divided by `nonKDim`

Reply:
tensor in MFMA layout (note: not dot operands, they are different) is composed of blocks of size `nonKDim` x `nonKDim`.
This block is a **minimal** size of a tensor in MFMA layout.

`shape[x] / nonKDim` is a number of these tiles over x dimension.

Note that in future there will be separate `mDim` and `nDim` instead of unified `nonKDim`

Reply:
I see, so currently `instr_shape` is expected to be equal on both dims, but in the future we will support non-equal mDim/nDim ? Does this also mean the tensor shape cannot be smaller than the instr_shape ?